content/developer/addons/events-and-handlers.md

```diff
@@ -109,3 +106,188 @@ With this addon enabled, going to the URL `/discussions/kaboom` would now output
 If you use a magic method to duplicate an existing method name, it will be overridden completely. And call to it will be directed to your plugin instead. The only exception is the `Index()` method.
 
 Magic methods only work in classes that extend `Gdn_Pluggable`. For example, notice the `Gdn_Form` class does, but the `Gdn_Format` class does not. All models and controllers do.
+
+## Example Events
+
+### Inject the the current user's roles into every page
+
+Sometimes you may want to adjust parts of the template based on the roles the current user. This will inject gather the roles of the current user and inject them into the smarty template.
+
+```php
+public function base_render_before($sender) {
+    if (inSection('Dashboard')) {
+        return;
+    }
+
+    if(!val('UserRoles', $sender->Data)) {
+        $user = val('User', Gdn::controller());
+        if (!$user && Gdn::session()->isValid()) {
+            $user = Gdn::session()->User;
+        }
+        $userRoles = [];
+        if ($user) {
+            $userID = val('UserID', $user);
+            $userRolesData = Gdn::userModel()->getRoles($userID);
+            if ($userRolesData->numRows() > 0) {
+                $userRoles = array_column($userRolesData->resultArray(), 'Name');
+            }
+        }
+        $sender->setData('UserRoles', $userRoles);
+    }
+}
+```
+
+### Inject a conditional based on roles
+
+Sometimes you may not need all the roles in the page. Let's say you wanted to make the page appear differently for a user with a certain role. Instead of injection all of the roles into the template and doing the conditional there, you can do it in the themehooks and inject just the boolean value you need.
+
+
+```php
+public function base_render_before($sender) {
+    if (inSection('Dashboard')) {
+        return;
+    }
+    
+    $roles = val('UserRoles', $sender->Data));
```

Comment:
Why is the code to fetch the roles different from the previous example?

Reply:
Corrected.